code/modules/power/supermatter/supermatter.dm

```diff
@@ -272,107 +272,106 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
+		//is on. An increase of 4*C @ 25% efficiency here results in an increase of 1*C / (#tilesincore) overall.
+		removed.temperature += ((device_energy * dynamic_heat_modifier) / THERMAL_RELEASE_MODIFIER)
 
-	var/device_energy = power * REACTION_POWER_MODIFIER
+		removed.temperature = max(0, min(removed.temperature, 2500 * dynamic_heat_modifier))
 
-	//To figure out how much temperature to add each tick, consider that at one atmosphere's worth
-	//of pure oxygen, with all four lasers firing at standard energy and no N2 present, at room temperature
-	//that the device energy is around 2140. At that stage, we don't want too much heat to be put out
-	//Since the core is effectively "cold"
+		//Calculate how much gas to release
+		removed.gases[/datum/gas/plasma][MOLES] += max((device_energy * dynamic_heat_modifier) / PLASMA_RELEASE_MODIFIER, 0)
 
-	//Also keep in mind we are only adding this temperature to (efficiency)% of the one tile the rock
-	//is on. An increase of 4*C @ 25% efficiency here results in an increase of 1*C / (#tilesincore) overall.
-	removed.temperature += ((device_energy * dynamic_heat_modifier) / THERMAL_RELEASE_MODIFIER)
+		removed.gases[/datum/gas/oxygen][MOLES] += max(((device_energy + removed.temperature * dynamic_heat_modifier) - T0C) / OXYGEN_RELEASE_MODIFIER, 0)
 
-	removed.temperature = max(0, min(removed.temperature, 2500 * dynamic_heat_modifier))
+		if(produces_gas)
+			env.merge(removed)
+			air_update_turf()
 
-	//Calculate how much gas to release
-	removed.gases[/datum/gas/plasma][MOLES] += max((device_energy * dynamic_heat_modifier) / PLASMA_RELEASE_MODIFIER, 0)
+		for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
+			if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
+				var/D = sqrt(1 / max(1, get_dist(l, src)))
+				l.hallucination += power * config_hallucination_power * D
+				l.hallucination = CLAMP(0, 200, l.hallucination)
 
-	removed.gases[/datum/gas/oxygen][MOLES] += max(((device_energy + removed.temperature * dynamic_heat_modifier) - T0C) / OXYGEN_RELEASE_MODIFIER, 0)
+		for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
+			var/rads = (power / 10) * sqrt( 1 / max(get_dist(l, src),1) )
+			l.rad_act(rads)
 
-	if(produces_gas)
-		env.merge(removed)
-		air_update_turf()
-
-	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
-			var/D = sqrt(1 / max(1, get_dist(l, src)))
-			l.hallucination += power * config_hallucination_power * D
-			l.hallucination = CLAMP(0, 200, l.hallucination)
-
-	for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
-		var/rads = (power / 10) * sqrt( 1 / max(get_dist(l, src),1) )
-		l.rad_act(rads)
+		power -= ((power/500)**3) * powerloss_inhibitor
```

Comment:
Assuming you make it still radiate and hallucinate on space, probably best to also deduct power on space